src/test/java/com/artipie/LayoutFlatTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import com.artipie.asto.Key;
+import com.artipie.repo.PathPattern;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link Layout.Flat}.
+ *
+ * @since 0.15
+ */
+final class LayoutFlatTest {
+
+    @Test
+    void hasNoDashboard() {
+        MatcherAssert.assertThat(
+            new Layout.Flat().hasDashboard(),
+            new IsEqual<>(false)
+        );
+    }
+
+    @Test
+    void usesFlatPattern() {
+        MatcherAssert.assertThat(
+            new Layout.Flat().pattern(),
+            new IsEqual<>(new PathPattern("flat").pattern())
+        );
+    }
+
+    @Test
+    void getKeyFromOnePartPath() {
+        MatcherAssert.assertThat(
+            new Layout.Flat().keyFromPath("/one").get(),
+            new IsEqual<>(new Key.From("one"))
+        );
+    }
```

Comment:
@baudoliver7 please also add test for path with more than one part. It was before but after changes you have deleted this test

Reply:
@genryxy good

Reply:
@genryxy Let me introduce a Parameterized Test :)

Reply:
@baudoliver7 yeah, let's try a parameterized test